androidshared/src/main/java/org/odk/collect/androidshared/system/ExternalFilesUtils.kt

```diff
@@ -8,13 +8,27 @@ object ExternalFilesUtils {
 
     @JvmStatic
     fun testExternalFilesAccess(context: Context) {
-        try {
-            val externalFilesDir = context.getExternalFilesDir(null)
-            val testFile = File(externalFilesDir.toString() + File.separator + ".test")
-            testFile.createNewFile()
-            testFile.delete()
-        } catch (e: IOException) {
-            throw IllegalStateException("App can't write to storage!")
+        val externalFilesDir = context.getExternalFilesDir(null)
+
+        if (externalFilesDir == null) {
+            throw IllegalStateException("External files dir is null!")
+        } else {
+            try {
+                val testFile = File(externalFilesDir.toString() + File.separator + ".test")
+                testFile.createNewFile()
+                testFile.delete()
+            } catch (e: IOException) {
+                if (!externalFilesDir.exists()) {
```

Comment:
This you could check earlier without waiting for exception.

Reply:
This is probably paranoid, but I think I'd rather have the exception occur first, so we know we're in a crash scenario instead of deciding to crash ourselves. I'd of course expect the dir not existing to cause a crash, but I'm also worried that some magic might allow it to get created on the fly when written to on some devices.

Reply:
ok